frontend-v2/src/app/users/user-table.tsx

```diff
@@ -0,0 +1,287 @@
+          <Button size="sm" variant="outline" asChild>
+            <Link href={`/users/${row.original.id}`}>Edit</Link>
+          </Button>
+        ),
+      },
+    ]
+  }, [chapterMap])
+
+  const table = useReactTable({
+    data: users,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    onSortingChange: setSorting,
+    state: {
+      sorting,
+    },
+  })
+
+  const rows = table.getRowModel().rows
+
+  return (
+    <div className="space-y-4">
+      <div className="rounded-md border hidden md:block">
+        <Table>
+          <TableHeader>
+            {table.getHeaderGroups().map((headerGroup) => (
+              <TableRow key={headerGroup.id}>
+                {headerGroup.headers.map((header) => (
+                  <TableHead key={header.id} className="whitespace-nowrap">
+                    {header.isPlaceholder
+                      ? null
+                      : flexRender(
+                          header.column.columnDef.header,
+                          header.getContext(),
+                        )}
+                  </TableHead>
+                ))}
+              </TableRow>
+            ))}
+          </TableHeader>
+          <TableBody>
+            {rows.length ? (
+              rows.map((row) => (
+                <TableRow key={row.id}>
+                  {row.getVisibleCells().map((cell) => (
+                    <TableCell key={cell.id}>
+                      {flexRender(
+                        cell.column.columnDef.cell,
+                        cell.getContext(),
+                      )}
+                    </TableCell>
+                  ))}
+                </TableRow>
+              ))
+            ) : (
+              <TableRow>
+                <TableCell
+                  colSpan={columns.length}
+                  className="text-center py-6"
+                >
+                  No users found.
+                </TableCell>
+              </TableRow>
+            )}
+          </TableBody>
+        </Table>
+      </div>
+
+      <div className="md:hidden space-y-3">
+        {rows.length ? (
+          rows.map((row) => (
+            <div
+              key={row.id}
+              className="rounded-lg border bg-card p-4 shadow-sm text-card-foreground"
+            >
+              <div className="flex items-start justify-between gap-3">
+                <div className="space-y-1">
+                  <Link
+                    href={`/users/${row.original.id}`}
+                    className="text-base font-semibold hover:underline"
+                  >
+                    {row.original.name}
+                  </Link>
+                  <span className="block text-sm text-muted-foreground">
+                    {row.original.email}
+                  </span>
+                </div>
+                <Button size="sm" variant="outline" asChild>
+                  <Link href={`/users/${row.original.id}`}>Edit</Link>
```

Comment:
how do you feel about having full pages to edit users vs. doing it in a modal?

i'm fine either way, and we can polish later. just curious how much thought you put into the UX of that.